lhotse/recipes/wenet_speech.py

```diff
@@ -0,0 +1,141 @@
+    assert corpus_dir.is_dir(), f"No such directory: {corpus_dir}"
+    if output_dir is not None:
+        output_dir = Path(output_dir)
+        output_dir.mkdir(parents=True, exist_ok=True)
+
+    subsets = WETNET_SPEECH_PARTS if "all" in dataset_parts else dataset_parts
+
+    manifests = defaultdict(dict)
+    for sub in subsets:
+        if sub not in WETNET_SPEECH_PARTS:
+            raise ValueError(f"No such part of dataset in WenetSpeech : {sub}")
+        manifests[sub] = {"recordings": [], "supervisions": []}
+
+    raw_manifests_path = corpus_dir / "WenetSpeech.json"
+    assert raw_manifests_path.is_file(), f"No such file : {raw_manifests_path}"
+    logging.info(f"Loading raw manifests from : {raw_manifests_path}")
+    raw_manifests = json.load(open(raw_manifests_path, "r", encoding="utf8"))
+
+    with ProcessPoolExecutor(num_jobs) as ex:
+        for recording, segments in tqdm(
+            ex.map(
+                parse_utterance,
+                raw_manifests["audios"],
+                repeat(corpus_dir),
+                repeat(subsets),
+            ),
+            desc="Processing WenetSpeech JSON entries",
+        ):
+            for part in segments:
+                manifests[part]["recordings"].append(recording)
+                manifests[part]["supervisions"].extend(segments[part])
+
+    for sub in subsets:
+        recordings, supervisions = fix_manifests(
+            recordings=RecordingSet.from_recordings(manifests[sub]["recordings"]),
+            supervisions=SupervisionSet.from_segments(manifests[sub]["supervisions"]),
+        )
+        validate_recordings_and_supervisions(
+            recordings=recordings, supervisions=supervisions
+        )
+
+        if output_dir is not None:
+            supervisions.to_json(output_dir / f"supervisions_{sub}.json")
+            recordings.to_json(output_dir / f"recordings_{sub}.json")
+
+        manifests[sub] = {
+            "recordings": recordings,
+            "supervisions": supervisions,
+        }
+
+    return manifests
+
+
+def parse_utterance(
+    audio: Any, root_path: Path, subsets: Sequence
+) -> Tuple[Recording, Dict[str, List[SupervisionSegment]]]:
+    sampling_rate = 16000
+    recording = Recording(
+        id=audio["aid"],
+        sources=[
+            AudioSource(
+                type="file",
+                channels=[0],
+                source=str(root_path / audio["path"]),
+            )
+        ],
+        num_samples=compute_num_samples(
+            duration=audio["duration"], sampling_rate=sampling_rate
+        ),
+        sampling_rate=sampling_rate,
+        duration=audio["duration"],
+    )
+    segments = defaultdict(dict)
+    for sub in subsets:
+        segments[sub] = []
+    for seg in audio["segments"]:
+        segment = SupervisionSegment(
+            id=seg["sid"],
+            recording_id=audio["aid"],
+            start=seg["begin_time"],
+            duration=seg["end_time"] - seg["begin_time"],
```

Comment:
I suggest to use `from lhotse.utils import add_durations` and then: 

```suggestion
            duration=add_durations(seg["end_time"], -seg["begin_time"], sampling_rate),
```

It will ensure that duration doesn't suffer from floating point precision errors.